src/test/java/io/confluent/connect/jdbc/sink/integration/PostgresViewIT.java

```diff
@@ -87,69 +91,131 @@ public void after() throws SQLException {
+        connect.kafka().produce(topic, null, kafkaValue);
+      } else {
+        kafkaValue = new String(jsonConverter.fromConnectData(topic, errorSchema, errorStruct));
+        connect.kafka().produce(topic, null, kafkaValue);
+      }
+    }
+
     try {
-      task.put(Collections.singleton(new SinkRecord(topic, 1, null, null, schema, struct, 1)));
-      fail();
-    } catch (ConnectException e) {
-      assertTrue(e.getMessage().contains("View \"" + topic + "\" is missing fields"));
+      // Consume the number of records produced since this is the upper bound of records that
+      // could be sent to the error reporter.
+      ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(6,
+          CONSUME_MAX_DURATION_MS, DLQ_TOPIC_NAME);
+    } catch (RuntimeException e) {
+      // Check that the amount of records that were actually found by the consumer matches the
+      // number of records expected to be sent to the error reporter.
+      assertEquals("3", e.toString().substring(65, 66));
     }
   }
 
   @Test
-  public void testRecordSchemaLessFieldsThanView() throws SQLException {
+  public void testRecordSchemaLessFieldsThanView() throws Exception {
     createTestTableAndView("firstName, lastName");
-    startTask();
+    connect.configureConnector("jdbc-sink-connector", props);
+    waitForConnectorToStart("jdbc-sink-connector", 1);
+
     final Schema schema = SchemaBuilder.struct().name("com.example.Person")
         .field("firstname", Schema.STRING_SCHEMA)
         .build();
     final Struct struct = new Struct(schema)
         .put("firstname", "Christina");
-    task.put(Collections.singleton(new SinkRecord(topic, 1, null, null, schema, struct, 1)));
-    try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
-      try (Statement s = c.createStatement()) {
-        try (ResultSet rs = s.executeQuery("SELECT * FROM " + topic)) {
-          assertTrue(rs.next());
-          assertEquals(struct.getString("firstname"), rs.getString("firstname"));
-          assertNull(rs.getString("lastname"));
-        }
-      }
-    }
+
+    String kafkaValue = new String(jsonConverter.fromConnectData(topic, schema, struct));
+    connect.kafka().produce(topic, null, kafkaValue);
+
+    waitForCondition(
+        () -> {
+          try (Connection c = pg.getEmbeddedPostgres().getPostgresDatabase().getConnection()) {
+            try (Statement s = c.createStatement()) {
+              try (ResultSet rs = s.executeQuery("SELECT * FROM " + topic)) {
+                boolean result = rs.next()
+                    && struct.getString("firstname").equals(rs.getString("firstname"))
+                    && rs.getString("lastname") == null;
+                return Optional.of(result).orElse(false);
+              }
+            }
+          }
+        },
+        VERIFY_MAX_DURATION_MS,
+        "The database content did not match the record's content."
+    );
```

Comment:
Same comment here w/r/t waiting for offsets and then performing a single database content assertion.